hummingbot/market/kucoin/kucoin_api_order_book_data_source.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python
+
+import asyncio
+import json
+import aiohttp
+import logging
+import pandas as pd
+from typing import (
+    Any,
+    AsyncIterable,
+    Dict,
+    List,
+    Optional
+)
+import re
+import time
+import ujson
+import websockets
+from websockets.exceptions import ConnectionClosed
+
+from hummingbot.core.utils import async_ttl_cache
+from hummingbot.core.utils.async_utils import safe_gather
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.data_type.order_book_tracker_entry import OrderBookTrackerEntry
+from hummingbot.core.data_type.order_book_message import OrderBookMessage
+from hummingbot.core.data_type.order_book import OrderBook
+from hummingbot.logger import HummingbotLogger
+from hummingbot.market.kucoin.kucoin_order_book import KucoinOrderBook
+
+TRADING_PAIR_FILTER = re.compile(r"(BTC|ETH|USDT)$")
+
+SNAPSHOT_REST_URL = "https://api.kucoin.com/api/v1/market/orderbook/level2_100"
```

Comment:
According to KuCoin's documentation, this endpoint only returns 
>  100 pieces of data (ask and bid data) on the order book 
I would recommend that you follow the instructions as stated in their documentation on maintaining a up-to-date Order Book. 

As stated [here](https://docs.kucoin.com/#level-2-market-data) they recommend that to maintain an order book, we use this `/api/v2/market/orderbook/level2` endpoint before applying the incremental market data from the websocket.

Reply:
Changed SNAPSHOT_REST_URL to /api/v2/market/orderbook/level2. Confirm in https://github.com/vic-en/hummingbot/blob/feat/kucoin-connector/hummingbot/market/kucoin/kucoin_api_order_book_data_source.py